prov/shm/src/smr_ep.c

```diff
@@ -410,25 +446,24 @@ static void smr_format_cma(struct smr_cmd *cmd, const struct fi_msg *msg,
 	cmd->hdr.op.rx_index = 0;
 	cmd->hdr.op.op = ofi_op_msg;
 	cmd->hdr.op.op_data = smr_src_iov;
-	cmd->hdr.op.flags = 0;
 
 	cmd->hdr.op.size = 0;
-	cmd->hdr.op.resv = 0;
 	resp->status = -FI_EBUSY;
 	cmd->hdr.op.data = (uint64_t) ((char **) resp - (char **) smr);
-	cmd->hdr.op.iov_count = msg->iov_count;
 
-	for (i = 0; i < msg->iov_count; i++) {
-		cmd->data.iov[i].iov_base = msg->msg_iov[i].iov_base;
-		cmd->data.iov[i].iov_len = msg->msg_iov[i].iov_len;
-		cmd->hdr.op.size += msg->msg_iov[i].iov_len;
+	for (i = 0; i < count; i++) {
+		cmd->data.iov[i].iov_base = iov[i].iov_base;
+		cmd->data.iov[i].iov_len = iov[i].iov_len;
+		cmd->hdr.op.size += iov[i].iov_len;
 	}
+	for (; i < SMR_IOV_LIMIT; cmd->data.iov[i++].iov_len = 0);
```

Comment:
nit: Please move ';' to next line to better indicate that this is an empty for-loop.  Or just convert to a while-loop